index.bs

```diff
@@ -52,72 +49,205 @@ RTCPeerConnection, canvas or MediaRecorder.
+<dd>Allows reading control signals sent from any sinks connected to the
+{{MediaStreamTrackGenerator}}. Control signals are objects of type {{MediaStreamTrackSignal}}.
+</dd>
+</dl>
+
+## Stream control ## {#stream-control}
+<pre class="idl">
+dictionary MediaStreamTrackSignal {
+  required MediaStreamSignalType signalType;
+  [EnforceRange] unsigned long long fps;
+}
 
-## Extension operation ## {#operation}
-
-
-### Stream creation ### {#stream-creation}
-
-
-### Stream processing ### {#stream-processing}
-
-### Stream control ### {#stream-control}
-
-The control signal set is platform-defined, but is intended to be extensible. By default, a
-TrackProcessor coupled with a TrackGenerator should be able to pass the signals unchanged
-to the other side.
+enum MediaStreamTrackSignalType {
+  "request-frame"
+  "set-min-fps",
+}
+</pre>
+In the MediaStream model, apart from media, which flows from sources to sinks, there are also
+control signals that flow in the opposite direction (i.e., from sinks to sources via the track).
+A {{MediaStreamTrackProcessor}} is a sink and it allows sending control
+signals to its track and source via its {{MediaStreamTrackProcessor/writableControl}} field.
+A {{MediaStreamTrackGenerator}} is a track for which a custom source can be implemented
+by writing media frames to its {{MediaStreamTrackProcessor/writable}} field. Such a source can
+receive control signals sent by sinks via its {{MediaStreamTrackProcessor/readableControl}} field.
+Note that control signals are just hints that a sink can send to its track (or the source
+backing it). There is no obligation for a source or track to react to them.
+
+Control signals are represented as {{MediaStreamTrackSignal}} objects.
+The {{MediaStreamTrackSignal}} dictionary has the following fields:
+* {{MediaStreamTrackSignal/signalType}}, which specifies the action to be requested to
+  the track or source. The possible values are defined by the {{MediaStreamTrackSignalType}}
+  enum and are the following:
+    * `"request-frame"`, tells the source to produce a new frame.
+    * `"set-min-fps"`, tells the source or track to maintain a
+      minimum frame rate.
+* {{MediaStreamTrackSignal/fps}} a numeric attribute representing a frame rate in frames per
+  second, used together with `"set-min-fps"` control signals.
+
+This set of control signals is intended to be extensible, so it is possible that new signal types
+and parameters may be added in the future. Note also that this set of control signals is
+not intended to cover all possible signaling that may occur between platform sinks and
+tracks/sources. A user agent implementation is free to implement any internal signaling between
+specific types of sinks and specific types of sources, and it would not make sense to expose all
+such specific signaling to generic Web platform objects like track generators and processors,
+as they can be considered implementation details that may differ significantly across browsers.
+
```

Comment:
Add something like: "It is, however, a requirement of this specification that it is possible to operate a MediaStreamTrackGenerator connected to a MediaStreamTrackProcessor using only the Web Platform-exposed signals and without connecting any implicit signalling."

Reply:
Done